internal/catalogmetadata/cache/cache.go

```diff
@@ -8,18 +8,21 @@ import (
 	"path/filepath"
 	"sync"
 
+	"github.com/go-logr/logr"
+
 	"github.com/operator-framework/operator-registry/alpha/declcfg"
 
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata/client"
 )
 
 var _ client.Cache = &filesystemCache{}
 
-func NewFilesystemCache(cachePath string) *filesystemCache {
+func NewFilesystemCache(cachePath string, logger logr.Logger) *filesystemCache {
 	return &filesystemCache{
 		cachePath:              cachePath,
 		mutex:                  sync.RWMutex{},
-		cacheDataByCatalogName: map[string]cacheData{},
+		cacheDataByCatalogName: sync.Map{},
```

Comment:
`sync.Map`'s lack of type safety is a concern. Could we hold off on this change until `sync/v2` package lands with a generic `sync.Map`?

Reply:
Also it looks like the existing mutex is still in place (for other reasons?), so the extra locking if the sync.Map doesn't buy us anything.